user/user.go

```diff
@@ -81,59 +93,86 @@ func (s *Store) GetUsers(ctx context.Context) ([]User, error) {
 func (s *Store) GetUsersSorted(ctx context.Context, column, direction string) ([]User, error) {
-	switch column {
-	case "userId":
-		switch direction {
-		case "asc":
-			return s.getUsersIDA(ctx)
-		case "desc":
-			return s.getUsersIDD(ctx)
-		}
-	case "name":
-		switch direction {
-		case "asc":
-			return s.getUsersFLNA(ctx)
-		case "desc":
-			return s.getUsersFLND(ctx)
-		}
-	case "username":
-		switch direction {
-		case "asc":
-			return s.getUsersUA(ctx)
-		case "desc":
-			return s.getUsersUD(ctx)
-		}
-	case "email":
-		switch direction {
-		case "asc":
-			return s.getUsersEA(ctx)
-		case "desc":
-			return s.getUsersED(ctx)
-		}
-	case "lastLogin":
-		switch direction {
-		case "asc":
-			return s.getUsersLLA(ctx)
-		case "desc":
-			return s.getUsersLLD(ctx)
-		}
+	switch direction {
+	case "asc":
+		return s.getUsersOptionsAsc(ctx, column)
+	case "desc":
+		return s.getUsersOptionsDesc(ctx, column)
 	}
 	return nil, fmt.Errorf("error in db sorting")
 }
 
+// GetUsersSortedSizePage returns a group of users, used for administration with sorting with size and page
+func (s *Store) GetUsersSortedSizePage(ctx context.Context, column, direction string, size, page int) ([]User, error) {
+	switch direction {
+	case "asc":
+		return s.getUsersOptionsAscSizePage(ctx, column, size, page)
+	case "desc":
+		return s.getUsersOptionsDescSizePage(ctx, column, size, page)
+	}
+	return nil, fmt.Errorf("error in db sorting size page")
+}
+
+// GetUsersSortedSearch returns a group of users, used for administration with sorting and searching
+func (s *Store) GetUsersSortedSearch(ctx context.Context, column, direction, search string) ([]User, error) {
+	switch direction {
+	case "asc":
+		return s.getUsersSearchOptionsAsc(ctx, search, column)
+	case "desc":
+		return s.getUsersSearchOptionsDesc(ctx, search, column)
+	}
+	return nil, fmt.Errorf("error in db sorting")
+}
+
+// GetUsersSortedSearchSizePage returns a group of users, used for administration with sorting and searching with pages
+func (s *Store) GetUsersSortedSearchSizePage(ctx context.Context, column, direction, search string, size, page int) ([]User, error) {
+	switch direction {
+	case "asc":
+		return s.getUsersSearchOptionsAscSizePage(ctx, search, column, size, page)
+	case "desc":
+		return s.getUsersSearchOptionsDescSizePage(ctx, search, column, size, page)
+	}
+	return nil, fmt.Errorf("error in db sorting size page")
+}
+
 // VerifyUser will check that the password is correct with provided
 // credentials and if verified will return the User object
-func (s *Store) VerifyUser(ctx context.Context, u User) (User, error) {
+func (s *Store) VerifyUser(ctx context.Context, u User) (User, bool, error) {
 	user, err := s.getUser(ctx, u)
 	if err != nil {
-		return u, fmt.Errorf("failed to get user: %w", err)
+		return u, false, fmt.Errorf("failed to get user: %w", err)
+	}
+	if !user.Enabled {
+		return u, false, errors.New("user not enabled, contact Computing Team for help")
```

Comment:
Possibly something to think about later down the line rather than in this (or subsequent) PRs, but it might be worth introducing a separate `error` type for errors that we're happy to show end-users - to ensure that users see "this user is disabled" but don't see internal SQL errors. Food for thought.